src/neuroconv/tools/nwb_helpers/_configure_backend.py

```diff
@@ -15,29 +15,64 @@ def configure_backend(
     """Configure all datasets specified in the `backend_configuration` with their appropriate DataIO and options."""
     nwbfile_objects = nwbfile.objects
 
+    # When a configuration is generated on an equivalent NWBFile elsewhere in memory, the object IDs will not match
+    # But a mapping can be formed to reconcile the two
+    # This is the case for the interface and converter usage of the feature (since they are separate workflow steps)
+    # As well as the NWB GUIDE (since they are separate endpoints in the Flask server)
+    nwbfile_object_ids = [nwbfile_object_id for nwbfile_object_id in nwbfile_objects.keys()]
+    dataset_configuration_object_ids = [
+        dataset_configuration.object_id in nwbfile_object_ids
+        for dataset_configuration in backend_configuration.dataset_configurations.values()
+    ]
+    dataset_configuration_to_nwbfile_object_ids = {  # Default to identity
+        dataset_configuration_object_id: dataset_configuration_object_id
+        for dataset_configuration_object_id in dataset_configuration_object_ids
+    }
+    if not all(
+        dataset_configuration_object_id in nwbfile_object_ids
+        for dataset_configuration_object_id in dataset_configuration_object_ids
+    ):
+        backend_configuration_class = type(backend_configuration)
+        new_default_backend_configuration = backend_configuration_class.from_nwbfile(nwbfile=nwbfile)
+        locations_in_file_to_new_object_ids = {
+            dataset_configuration.location_in_file: dataset_configuration.object_id
+            for dataset_configuration in new_default_backend_configuration.dataset_configurations.values()
+        }
+
+        dataset_configuration_to_nwbfile_object_ids = {
+            dataset_configuration.object_id: locations_in_file_to_new_object_ids[dataset_configuration.location_in_file]
+            for dataset_configuration in backend_configuration.dataset_configurations.values()
+        }
+
+    # Set all DataIO based on the configuration
     data_io_class = backend_configuration.data_io_class
     for dataset_configuration in backend_configuration.dataset_configurations.values():
-        object_id = dataset_configuration.object_id
+        object_id = dataset_configuration_to_nwbfile_object_ids[dataset_configuration.object_id]
         dataset_name = dataset_configuration.dataset_name
         data_io_kwargs = dataset_configuration.get_data_io_kwargs()
 
         # TODO: update buffer shape in iterator, if present
 
-        nwbfile_object = nwbfile_objects[object_id]
-        is_dataset_linked = isinstance(nwbfile_object.fields.get(dataset_name), TimeSeries)
+        neurodata_object = nwbfile_objects[object_id]
```

Comment:
Noticed that the name in the other models (the ones actually exposed) are `neurodata_type` so changed this for consistency

I think all `nwbfile.objects` are technically 'neurodata objects' (instances of neurodata types) by definition, but whatevs

Reply:
This makes sense to me. I wish it was in a different PR to make the diff simpler but your time is also valuable.

Reply:
Opened https://github.com/catalystneuro/neuroconv/pull/817